oci/pull.bzl

```diff
@@ -0,0 +1,99 @@
+"Pull image layers using Bazel downloader"
+
+_attrs = {
+    "image": attr.string(),
+    "tag": attr.string(),
+}
+
+def _download(rctx, tag, output, type = "manifests"):
+    if type != "blobs" and type != "manifests":
+        fail("type must be blobs or manifests")
+    if rctx.attr.image.startswith("gcr.io/"):
+        registry_url = "https://gcr.io/v2/{image}/{type}/{tag}".format(
+            image = rctx.attr.image[len("gcr.io/"):],
+            type = type,
+            tag = tag,
+        )
+    else:
+        fail("Unrecognized registry. We only understand gcr.io currently. File an issue on rules_oci.")
+
+    sha256 = None
+    if tag.startswith("sha256:"):
+        sha256 = tag[len("sha256:"):]
+    rctx.download(
+        url = registry_url,
+        sha256 = sha256,
+        output = output,
+    )
+    if type == "manifests":
+        return json.decode(rctx.read(output))
+    return None
+
+def _download_blobs(rctx, tag, output):
+    _download(rctx, tag, output, "blobs")
+
+_build_file_header = """\
+# Generated by oci_pull
+
+load("@contrib_rules_oci//oci:defs.bzl", "oci_image")
+
+package(default_visibility = ["//visibility:public"])
+"""
+
+_oci_image_target = """\
+oci_image(
```

Comment:
nit: instead of `crane manifest gcr.io/distroless/static:latest --platform linux/arm64` and only getting one manifest, we may as well ask the user to run `crane manifest gcr.io/distroless/static:latest` and then we have all platforms

Reply:
I don't have any ideas how to assemble a base image from the manifest and layers, maybe we need another crane command that's like `pull` but works offline?

Reply:
Using `copy_to_directory` it's easy to create the same layout as `crane pull` - the only problem is I can't find any endpoint on the remote registry that serves the "config" file, even though we have its digest.

Reply:
for offline use cases, crane pull wouldn't work cause it has to hit the registry in order to do some calculations.

Reply:
what kind of calculations? I'm not doing those here, just constructing an oci layout folder like crane does.

Reply:
it'll hit the remote registry to see what a tag refers to for instance if the reference has tags. in most cases, it will try to talk to the remote registries to fetch manifests to see what they consist of.

Reply:
> just constructing an oci layout folder like crane does.

this is good enough. as long as it's a valid oci layout that contains a valid manifest and config alongside blobs

Reply:
I added a test that runs crane pull and asserts that it produces an identical `layout` folder to what this repo rule produces.